### PR TITLE
Use the X.509 typography instead of "x509"

### PR DESCRIPTION
**Pull request by maelvls**
Looks like we started using the "x509" typography everywhere, but shouldn't we use "X.509" instead as started in [RFC 5280](https://tools.ietf.org/html/rfc5280) 😅

At first, I went with a quick PCRE which replaces all occurrences of `x509` except when it is used in the context of `openssl x509`:

```sh
perl -pi~ -n -e 's/(?<!openssl )x509/X.509/g' content/en/**/*.md && rm content/en/**/*.md~
```

But then I released that this change only fixes the spelling for the current documentation (i.e., master), not for all the other release branches from which we generate some parts of the website:

| branch       | spelling is fixed? | PR fixing the spelling |
| ------------ | :----------------: | ---------------------- |
| master       |         ✅          | #453                   |
| release-next |         ❌          |                        |
| release-1.2  |         ❌          |                        |
| release-1.1  |         ❌          |                        |
| release-0.12 |         ❌          |                        |
| release-0.13 |         ❌          |                        |
| release-0.14 |         ❌          |                        |
| release-0.15 |         ❌          |                        |
| release-0.16 |         ❌          |                        |

So I ignored all the above non-fixed branches with the following in `.spelling`:

```sh
# As per https://tools.ietf.org/html/rfc5280, "X.509" is the correct spelling.
# The spellings "x509" and "X509" are misspellings.
X.509

# Since the Markdown files in content/en/*-docs are copied over from the
# cert-manager repository using fixed tags, we can't expect these files to
# contain the right X.509 spelling. The following is a series of exceptions
# so that the x509 spelling is accepted in those files.
 - content/en/next-docs/concepts/certificate.md
x509
 - content/en/next-docs/concepts/certificaterequest.md
x509
 - content/en/next-docs/concepts/webhook.md
x509
 - content/en/next-docs/release-notes/release-notes-0.13.md
x509
 - content/en/next-docs/release-notes/release-notes-0.15.md
x509
 - content/en/next-docs/release-notes/release-notes-0.16.md
x509
 - content/en/next-docs/release-notes/release-notes-0.5.md
x509
 - content/en/next-docs/release-notes/release-notes-0.6.md
x509
 - content/en/next-docs/release-notes/release-notes-0.7.md
x509
 - content/en/next-docs/release-notes/release-notes-0.9.md
x509
 - content/en/next-docs/tutorials/venafi/venafi.md
x509
 - content/en/next-docs/usage/certificate.md
x509
 - content/en/next-docs/usage/kubectl-plugin.md
x509
 - content/en/v0.12-docs/concepts/certificate.md
x509
 - content/en/v0.12-docs/concepts/certificaterequest.md
x509
 - content/en/v0.12-docs/release-notes/release-notes-0.5.md
x509
 - content/en/v0.12-docs/release-notes/release-notes-0.6.md
x509
 - content/en/v0.12-docs/release-notes/release-notes-0.7.md
x509
 - content/en/v0.12-docs/release-notes/release-notes-0.9.md
x509
 - content/en/v0.12-docs/tutorials/venafi/venafi.md
x509
 - content/en/v0.12-docs/usage/certificate.md
x509
 - content/en/v0.13-docs/concepts/certificate.md
x509
 - content/en/v0.13-docs/concepts/certificaterequest.md
x509
 - content/en/v0.13-docs/release-notes/release-notes-0.13.md
x509
 - content/en/v0.13-docs/release-notes/release-notes-0.5.md
x509
 - content/en/v0.13-docs/release-notes/release-notes-0.6.md
x509
 - content/en/v0.13-docs/release-notes/release-notes-0.7.md
x509
 - content/en/v0.13-docs/release-notes/release-notes-0.9.md
x509
 - content/en/v0.13-docs/tutorials/venafi/venafi.md
x509
 - content/en/v0.13-docs/usage/certificate.md
x509
 - content/en/v0.14-docs/concepts/certificate.md
x509
 - content/en/v0.14-docs/concepts/certificaterequest.md
x509
 - content/en/v0.14-docs/release-notes/release-notes-0.13.md
x509
 - content/en/v0.14-docs/release-notes/release-notes-0.5.md
x509
 - content/en/v0.14-docs/release-notes/release-notes-0.6.md
x509
 - content/en/v0.14-docs/release-notes/release-notes-0.7.md
x509
 - content/en/v0.14-docs/release-notes/release-notes-0.9.md
x509
 - content/en/v0.14-docs/tutorials/venafi/venafi.md
x509
 - content/en/v0.14-docs/usage/certificate.md
x509
 - content/en/v0.15-docs/concepts/certificate.md
x509
 - content/en/v0.15-docs/concepts/certificaterequest.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.13.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.15.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.5.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.6.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.7.md
x509
 - content/en/v0.15-docs/release-notes/release-notes-0.9.md
x509
 - content/en/v0.15-docs/tutorials/venafi/venafi.md
x509
 - content/en/v0.15-docs/usage/certificate.md
x509
 - content/en/v0.16-docs/concepts/certificate.md
x509
 - content/en/v0.16-docs/concepts/certificaterequest.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.13.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.15.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.16.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.5.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.6.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.7.md
x509
 - content/en/v0.16-docs/release-notes/release-notes-0.9.md
x509
 - content/en/v0.16-docs/tutorials/venafi/venafi.md
x509
 - content/en/v0.16-docs/usage/certificate.md
x509
 - content/en/v0.16-docs/usage/kubectl-plugin.md
x509
```

In order to be able to remove all the above exceptions, we have to fix the spelling in the individual branches. The current PR only focus on master.

### CODE REVIEW

**Comment by maelvls** [comment]
Since I could not find a proper way of fixing `.spelling` so that it would only check spelling for non-generated documentation, I went with a quick PCRE:

```sh
perl -pi~ -n -e 's/(?<!openssl )x509/X.509/g' content/en/**/*.md && rm content/en/**/*.md~
```

It replaces all occurrences of `x509` except when it is used in the context of `openssl x509`.

File: .spelling [hunk 1]
```diff
@@ -209,3 +208,136 @@ ArtifactHub
 CryptoKey
 Encrypter
 cmrel
+
+# As per the RFCs, X.509 is the right spelling. x509 and X509 are
```
Comment (irbekrm) [inline .spelling:212]: Suggestion: perhaps we could link the relevant RFC here?

**Review by irbekrm — APPROVED** [review]
Looks great to me 🥳 

One minor suggestion, feel free to ignore that.

File: .spelling (hunk 1 above)
Comment (maelvls) [reply .spelling]: Good idea; I rephreased and added a link to https://tools.ietf.org/html/rfc5280 👍

**Pull request merged**